main/lsp/lsp.h

```diff
@@ -169,18 +180,22 @@ class LSPLoop {
         std::vector<std::unique_ptr<core::lsp::QueryResponse>> responses;
         // The global state, post-typechecking.
         std::unique_ptr<core::GlobalState> gs;
+        // The edit applied to `gs`.
+        std::optional<LSPLoop::FileUpdates> updates;
         bool tookFastPath;
     };
     /** Conservatively rerun entire pipeline without caching any trees */
-    TypecheckRun runSlowPath();
+    TypecheckRun runSlowPath(std::optional<FileUpdates> updates,
```

Comment:
does it have to take `optional`? can it just take `FileUpdates` where all vectors are empty?

Reply:
(ditto in TypecheckRun)